From 00197239142c519270e44f94b1126a45e7f65511 Mon Sep 17 00:00:00 2001
From: Michael Marineau <michael.marineau@coreos.com>
Date: Sat, 2 Aug 2014 17:45:00 -0700
Subject: [PATCH 35/37] hack: testing -Wl,-fuse-ld=gold does not work correctly
 on hardened

Not sure why this test falsely passes with the hardened compiler when it
normally will report the following error:

    ld: -f may not be used without -shared

But apparently the default options hardened uses makes interpreting the
option as -f valid usage. For reference the option is:

    -f name
    --auxiliary=name
        When creating an ELF shared object, set the internal DT_AUXILIARY
        field to the specified name.  This tells the dynamic linker that
        the symbol table of the shared object should be used as an
        auxiliary filter on the symbol table of the shared object name.

This in turn causes a stray library to show up in ldd output:

    use-ld=gold => not found

Which seems mostly harmless but does cause some confusion.
---
 configure.ac | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index ae88382..85966b9 100644
--- a/configure.ac
+++ b/configure.ac
@@ -216,8 +216,7 @@ CC_CHECK_FLAGS_APPEND([with_ldflags], [LDFLAGS], [\
         -Wl,--gc-sections \
         -Wl,-z,relro \
         -Wl,-z,now \
-        -pie \
-        -Wl,-fuse-ld=gold])
+        -pie])
 AC_SUBST([OUR_LDFLAGS], "$with_ldflags $sanitizer_ldflags")
 
 AC_CHECK_SIZEOF(pid_t)
-- 
1.8.5.5

